price-feeder/oracle/client/client.go

```diff
@@ -159,6 +173,8 @@ func (oc OracleClient) BroadcastTx(nextBlockHeight, timeoutHeight int64, msgs ..
 				Str("tx_hash", hash).
 				Uint32("tx_code", code).
 				Msg("failed to broadcast tx; retrying...")
+
+			time.Sleep(time.Second * 1)
```

Comment:
Since we already subscribe to the blocks, why don't we refactor and use that information here as well? Do we need to send new price every second, or it's enough to do it once per block?

Reply:
actually, this sleep is to avoid the `for` to ask that much the chain height (even if it is cached, I don't see the point of spending process to keep getting the current height)

Reply:
I see, I didn't unroll the whole function body. Should we have max tries instead of max height?

Reply:
@adamewozniak what do you think?